cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -1589,19 +1614,16 @@ def validate_tags(self, tags):
         validate_no_reserved_tag(tags)
 
     @validates_schema
-    def no_intel_software_for_batch(self, data, **kwargs):
-        """Ensure IntelSoftware section is not included when AWS Batch is the scheduler."""
+    def no_settings_for_batch(self, data, **kwargs):
+        """Ensure IntelSoftware and DirectoryService section is not included when AWS Batch is the scheduler."""
         scheduling = data.get("scheduling")
-        additional_packages = data.get("additional_packages")
-        if (
-            scheduling
-            and scheduling.scheduler == "awsbatch"
-            and additional_packages
-            and additional_packages.intel_software.intel_hpc_platform
-        ):
-            raise ValidationError(
-                "The use of the IntelSoftware configuration is not supported when using awsbatch as the scheduler."
-            )
+        if scheduling and scheduling.scheduler == "awsbatch":
+            error_message = "The use of the {} configuration is not supported when using awsbatch as the scheduler."
+            additional_packages = data.get("additional_packages")
+            if additional_packages and additional_packages.intel_software.intel_hpc_platform:
+                raise ValidationError(error_message.format("IntelSoftware"))
+            if data.get("directory_service"):
+                raise ValidationError(error_message.format("DirectoryService"))
```

Comment:
this is a validation error that should be raised when either BYOS or Batch are used for the scheduler. For that reason I don't think it should be combined with this validator.

Reply:
actually i think i was wrong here.